inference_overview.md

```diff
@@ -0,0 +1,450 @@
+  invocation.
+1. The calculated bid is returned. To compute the bid, `generateBid()` can run
+  models through `runInference()calls` and combine the output embedding with
+  other embeddings that were fetched above to make a final prediction.
+
+In the above flow, `prepareDataForAdRetrieval()` and `generateBid()` can use
+external models through the `runInference()` API call. This runs inside the TEE
+and can perform inference with models using private features (e.g. the user
+tower). Alternatively (not shown in figure), each of the UDFs can use embedded
+models to perform inference.
+
+## Versioning
+
+Inference on B&A will support factorized models. There are multiple models and
+embeddings that contribute towards the final prediction, some of which may be
+pre-computed. It is important to make sure that model and embedding versions
+used are consistent.
+
+For facilitating this, we will provide the buyer a way to choose the version of
+models to run per request. As a part of contextual request processing, a version
+for inference can be decided by the buyer. The model version information can be
+passed through the `buyer_signals`. This field is a string and can be used to
+pass arbitrary information from the contextual RTB request to `generateBid().
+buyer_signals` will also be passed to the K/V server where this model version
+information can be used to look up embeddings. The ad tech is free to choose the
+structure of the `buyer_signals` including how the model information is passed
+and what it contains. For example, something like the following could be used:
+
+### Using versions in B&A
+
+The `runInference()` API runs inferences against a named model. `generateBid()`
+is responsible for interpreting the version included in `buyer_signals` and
+translating it to an actual path that corresponds to an entity in the cloud
+bucket. A simple translation scheme could be something like
+`${model_name}-${model_version`}. This name will be translated into a cloud
+bucket folder from which the model was loaded as shown in the figure below. This
+figure is for illustrative purposes only. It shows the mapping from model path
+to an actual path in the cloud bucket. The cloud bucket will only be synced
+periodically, so each `runInference()` call will not call out to the cloud
+bucket. Thus, there will be some delay between a new model being available in
+the cloud bucket, and it being made available to `runInference()`. We will cover
+such details in future updates to this explainer.
+
+<figure id = "image-4">
+  <img src = "images/inference_overview_fig4.png"
+  width = "100%"
+  alt = "Model name translation">
+  <figcaption><b>Figure 4.</b> Model name translation </figcaption>
+</figure>
+<br>
+
+The `getModelPaths()` function described [earlier][3] can be used along with
+the version information to implement a versioning strategy by calling
+`runInference()` with the chosen version. Here are some examples how this could
+be done:
```

Comment:
Where are the examples? Please double check.

Reply:
Oh right, we removed the examples. Updated.

Reply:
I do not see it yet. Please commit the change.